sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceUtils.scala

```diff
@@ -60,8 +60,8 @@ object DataSourceUtils {
   // SPARK-24626: Metadata files and temporary files should not be
   // counted as data files, so that they shouldn't participate in tasks like
   // location size calculation.
-  private[sql] def isDataPath(path: Path): Boolean = {
-    val name = path.getName
-    !(name.startsWith("_") || name.startsWith("."))
-  }
+  private[sql] def isDataPath(path: Path): Boolean = isDataFile(path.getName)
+
+  private[sql] def isDataFile(fileName: String) =
+    !(fileName.startsWith("_") || fileName.startsWith("."))
```

Comment:
This is actually different from our code which is calling `fs.getContentSummary(tablePath).getLength` in https://github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala#L125 

getContentSummary is using isChecksumFile to filter out the metadata file. 

https://hadoop.apache.org/docs/r2.7.4/api/src-html/org/apache/hadoop/fs/ChecksumFileSystem.html

```Java
093  /** Return true iff file is a checksum file name.*/
094  public static boolean isChecksumFile(Path file) {
095    String name = file.getName();
096    return name.startsWith(".") && name.endsWith(".crc");
097  }
```

Obviously, they are different.